js/models/conversations.js

```diff
@@ -779,14 +790,73 @@
                   var sessionCipher = new libsignal.SessionCipher(textsecure.storage.protocol, address);
                   return sessionCipher.closeOpenSessionForDevice();
               }
-            });
-        }).catch(function(error) {
+            }).then(function() {
+              var c = ConversationController.get(id);
+              return Promise.all([
+                c.setProfileName(profile.name),
+                c.setProfileAvatar(profile.avatar)
+              ]).then(function() {
+                // success
+                c.save();
```

Comment:
Probably makes sense to surround this with a `new Promise()` to capture any errors generated.

Reply:
done